org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLWorkspaceService.java

```diff
@@ -27,8 +30,14 @@
  */
 public class XMLWorkspaceService implements WorkspaceService {
 
-	private final XMLLanguageServer xmlLanguageServer;
+	public interface ExecuteCommandHandler {
+		CompletableFuture<Object> handle(ExecuteCommandParams params);
+	}
 
+	private final XMLLanguageServer xmlLanguageServer;
+	
+	private final Map<String, ExecuteCommandHandler> commands = new HashMap<>();
+	
 	public XMLWorkspaceService(XMLLanguageServer xmlLanguageServer) {
 		this.xmlLanguageServer = xmlLanguageServer;

```

Comment:
Initialize commands = new HashMap<>(); in the constructor.

Reply:
Will do, sounds good